File: go/app/acc/internal/api/turn.go
//
// weproxy@foxmail.com 2022/10/29
//

package api

import (
	"io"

	"weproxy/acc/libgo/logx"
)

// newTurnCli ...
func newTurnCli() (io.Closer, error) {
	m := &turnCli{}
	if err := m.Start(); err != nil {
		return nil, err
	}
	return m, nil
}

// turnCli ...
type turnCli struct {
}

// Start ...
func (m *turnCli) Start() error {
	logx.D("%s turnCli.Start()", TAG)
	return nil
}

// Close ...
func (m *turnCli) Close() error {
	logx.D("%s turnCli.Close()", TAG)
	return nil
}

File: cc/app/acc/internal/api/turn.cc
//
// weproxy@foxmail.com 2022/10/03
//

#include "api.h"
#include "gx/io/io.h"
#include "logx/logx.h"

namespace internal {
namespace api {

// turnCli ...
struct turnCli : public io::xx::closer_t {
    // Start ...
    error Start() {
        LOGF_D("%s turnCli.Start()", TAG);
        return nil;
    }

    // Close ...
    error Close() {
        LOGF_D("%s turnCli.Close()", TAG);
        return nil;
    }
};

// newTurnCli ...
R<io::Closer, error> newTurnCli() {
    auto m = NewRef<turnCli>();
    auto err = m->Start();
    if (err != nil) {
        return {nil, err};
    }
    return {m, nil};
}

}  // namespace api
}  // namespace internal
